# Tired of Cliché Datasets? Here Are 20 Awesome Alternatives For Any Task
## TODO
![](images/unsplash.jpg)
<figcaption style="text-align: center;">
    <strong>
        Photo by 
        <a href='https://unsplash.com/@dogukan?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'> Doğukan Şahin</a>
        on 
        <a href='https://unsplash.com/s/photos/tired?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'>Pexels.</a> All images are by the author unless specified otherwise.
    </strong>
</figcaption>

### Setup

In [7]:
import warnings

import gdown
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns
from matplotlib import rcParams

rcParams["xtick.labelsize"] = 15
rcParams["ytick.labelsize"] = 15

warnings.filterwarnings("ignore")

"I'm going to puke over my RGB backlit-keyboard *so hard* if I see one more person using Titanic, Iris, Wine or Boston datasets!"

This is the feeling you might gradually develop after being a data science learner for a while. You just can't help it - everyone wants the easy thing. Beginners use these datasets because they are stupidly straightforwad; most course creaters and bloggers use them because they are just one single Google search away.

# Regression datasets

## 1. Diamond prices and carat regression

My favorite from this list is the diamonds dataset. It is ideal in length (+50k samples) and have multiple targets you can predict as a regression or a multi-class classification task:

In [4]:
diamonds = sns.load_dataset("diamonds").sample(10).reset_index(drop=True)
diamonds

carat        cut color clarity  depth  table  price     x     y     z
0   0.90  Very Good     F     VS1   61.2   56.0   5324  6.18  6.23  3.80
1   0.41       Good     J     VS2   64.0   59.0    732  4.71  4.64  2.99
2   0.92  Very Good     H     VS1   62.9   58.0   4228  6.17  6.20  3.89
3   0.24    Premium     H    VVS1   61.2   58.0    432  3.96  4.01  2.44
4   0.30    Premium     E     VS2   59.2   59.0    844  4.41  4.37  2.60
5   0.93  Very Good     F     VS2   62.7   58.0   4925  6.21  6.27  3.91
6   0.41    Premium     J    VVS1   62.2   59.0    994  4.75  4.71  2.94
7   0.41    Premium     G     VS1   60.5   58.0    899  4.76  4.80  2.89
8   0.30      Ideal     G    VVS1   62.0   55.0   1013  4.34  4.31  2.68
9   0.71  Very Good     H     VS2   61.8   58.0   2368  5.66  5.70  3.51

In [5]:
sns.load_dataset("diamonds").shape

(53940, 10)

**🎯 Targets: 'carat', 'price' (regression); 'cut', 'color', 'clarity' (multi-class)**

**🔗 Link: [Kaggle](https://www.kaggle.com/shivam2503/diamondshttps://www.kaggle.com/shivam2503/diamonds)**

**📦Dimensions: (53940, 10)**

**⚙Missing values: No**

**[📚Starter notebook](https://www.kaggle.com/fuzzywizard/diamonds-in-depth-analysishttps://www.kaggle.com/fuzzywizard/diamonds-in-depth-analysis)**

## 2. Age of Abalone shells

This is a rather unique dataset from the field of zoology. The task is to predict the age of Abalone shells (a type of mollusc) using several physical measurements. Traditionally, their age are found by cutting through their cone, staining them and counting the number of rings inside the shell under a microscope. 

For zooligists, this might be fun but for data scientists, not so much:

In [12]:
abalone = pd.read_csv("data/reg6_abalone.csv")
abalone.head(10)

Sex  Length  Diameter  Height  Whole weight  Shucked weight  Viscera weight  \
0   M   0.455     0.365   0.095        0.5140          0.2245          0.1010   
1   M   0.350     0.265   0.090        0.2255          0.0995          0.0485   
2   F   0.530     0.420   0.135        0.6770          0.2565          0.1415   
3   M   0.440     0.365   0.125        0.5160          0.2155          0.1140   
4   I   0.330     0.255   0.080        0.2050          0.0895          0.0395   
5   I   0.425     0.300   0.095        0.3515          0.1410          0.0775   
6   F   0.530     0.415   0.150        0.7775          0.2370          0.1415   
7   F   0.545     0.425   0.125        0.7680          0.2940          0.1495   
8   M   0.475     0.370   0.125        0.5095          0.2165          0.1125   
9   F   0.550     0.440   0.150        0.8945          0.3145          0.1510   

   Shell weight  Rings  
0         0.150     15  
1         0.070      7  
2         0.210      9  
3         0.155     10  
4         0.055      7  
5         0.120      8  
6         0.330     20  
7         0.260     16  
8         0.165      9  
9         0.320     19

**🎯 Target: 'Rings'**

**🔗 Link: [Kaggle](https://www.kaggle.com/rodolfomendes/abalone-dataset)**

**📦Dimensions: (4177, 9)**

**⚙Missing values: No**

**[📚Starter notebook](https://www.kaggle.com/ragnisah/eda-abalone-age-prediction)**

## 3. King county house sales

This is the dataset for those who are still interested in real estate and house prices regression:

In [15]:
king = pd.read_csv("data/reg1_house_sales_king_county.csv")
king.head(10)

price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  waterfront  \
0   221900.0       3.0       1.00         1180      5650     1.0           0   
1   538000.0       3.0       2.25         2570      7242     2.0           0   
2   180000.0       2.0       1.00          770     10000     1.0           0   
3   604000.0       4.0       3.00         1960      5000     1.0           0   
4   510000.0       3.0       2.00         1680      8080     1.0           0   
5  1225000.0       4.0       4.50         5420    101930     1.0           0   
6   257500.0       3.0       2.25         1715      6819     2.0           0   
7   291850.0       3.0       1.50         1060      9711     1.0           0   
8   229500.0       3.0       1.00         1780      7470     1.0           0   
9   323000.0       3.0       2.50         1890      6560     2.0           0   

   view  condition  grade  sqft_above  yr_built  yr_renovated      lat  \
0     0          3      7        1180      1955           NaN  47.5112   
1     0          3      7        2170      1951        1991.0  47.7210   
2     0          3      6         770      1933           NaN  47.7379   
3     0          5      7        1050      1965           NaN  47.5208   
4     0          3      8        1680      1987           NaN  47.6168   
5     0          3     11        3890      2001           NaN  47.6561   
6     0          3      7        1715      1995           NaN  47.3097   
7     0          3      7        1060      1963           NaN  47.4095   
8     0          3      7        1050      1960           NaN  47.5123   
9     0          3      7        1890      2003           NaN  47.3684   

      long  sqft_living15  sqft_lot15  
0 -122.257           1340        5650  
1 -122.319           1690        7639  
2 -122.233           2720        8062  
3 -122.393           1360        5000  
4 -122.045           1800        7503  
5 -122.005           4760      101930  
6 -122.327           2238        6819  
7 -122.315           1650        9711  
8 -122.337           1780        8113  
9 -122.031           2390        7570

In [16]:
king.shape

(21613, 17)

**🎯 Target: 'price'**

**🔗 Link: [Kaggle](https://www.kaggle.com/harlfoxem/housesalesprediction)**

**📦Dimensions: (21613, 17)**

**⚙Missing values: Yes**

**[📚Starter notebook](https://www.kaggle.com/burhanykiyakoglu/predicting-house-prices)**

## 4. Cancer death rate

This dataset challenges you to find cancer mortality rate per capita (100,000) using a number of demographic variables:

In [18]:
cancer = pd.read_csv("data/reg3_cancer_death_rate.csv")
cancer.head(10)

avgAnnCount  avgDeathsPerYear  TARGET_deathRate  incidenceRate  medIncome  \
0       1397.0               469             164.9          489.8      61898   
1        173.0                70             161.3          411.6      48127   
2        102.0                50             174.7          349.7      49348   
3        427.0               202             194.8          430.4      44243   
4         57.0                26             144.4          350.1      49955   
5        428.0               152             176.0          505.4      52313   
6        250.0                97             175.9          461.8      37782   
7        146.0                71             183.6          404.0      40189   
8         88.0                36             190.5          459.4      42579   
9       4025.0              1380             177.8          510.9      60397   

   popEst2015  povertyPercent  studyPerCap           binnedInc  MedianAge  \
0      260131            11.2   499.748204   (61494.5, 125635]       39.3   
1       43269            18.6    23.111234  (48021.6, 51046.4]       33.0   
2       21026            14.6    47.560164  (48021.6, 51046.4]       45.0   
3       75882            17.1   342.637253    (42724.4, 45201]       42.8   
4       10321            12.5     0.000000  (48021.6, 51046.4]       48.3   
5       61023            15.6   180.259902  (51046.4, 54545.6]       45.4   
6       41516            23.2     0.000000  (37413.8, 40362.7]       42.6   
7       20848            17.8     0.000000  (37413.8, 40362.7]       51.7   
8       13088            22.3     0.000000  (40362.7, 42724.4]       49.3   
9      843954            13.1   427.748432  (54545.6, 61494.5]       35.8   

   ...  PctPrivateCoverageAlone  PctEmpPrivCoverage  PctPublicCoverage  \
0  ...                      NaN                41.6               32.9   
1  ...                     53.8                43.6               31.1   
2  ...                     43.5                34.9               42.1   
3  ...                     40.3                35.0               45.3   
4  ...                     43.9                35.1               44.0   
5  ...                     38.8                32.6               43.2   
6  ...                     35.0                28.3               46.4   
7  ...                     33.1                25.9               50.9   
8  ...                     37.8                29.9               48.1   
9  ...                      NaN                44.4               31.4   

   PctPublicCoverageAlone   PctWhite  PctBlack  PctAsian  PctOtherRace  \
0                    14.0  81.780529  2.594728  4.821857      1.843479   
1                    15.3  89.228509  0.969102  2.246233      3.741352   
2                    21.1  90.922190  0.739673  0.465898      2.747358   
3                    25.0  91.744686  0.782626  1.161359      1.362643   
4                    22.7  94.104024  0.270192  0.665830      0.492135   
5                    20.2  84.882631  1.653205  1.538057      3.314635   
6                    28.7  75.106455  0.616955  0.866157      8.356721   
7                    24.1  89.406636  0.305159  1.889077      2.286268   
8                    26.6  91.787477  0.185071  0.208205      0.616903   
9                    16.5  74.729668  6.710854  6.041472      2.699184   

   PctMarriedHouseholds  BirthRate  
0             52.856076   6.118831  
1             45.372500   4.333096  
2             54.444868   3.729488  
3             51.021514   4.603841  
4             54.027460   6.796657  
5             51.220360   4.964476  
6             51.013900   4.204317  
7             48.967033   5.889179  
8             53.446998   5.587583  
9             50.063573   5.533430  

[10 rows x 33 columns]

**🎯 Target: 'TARGET_deathRate'**

**🔗 Link: [Data.world](https://data.world/nrippner/ols-regression-challenge)**

**📦Dimensions: (3047, 33)**

**⚙Missing values: Yes**